### DISCUSSION TITLE
Support for ES Modules

### DISCUSSION THREAD

**Discussion by panuchka** [geckosio/geckos.io#92] [Ideas]
Would it be possible to add ESM support for the Geckos.io library? 

As JavaScript world is (slowly) moving towards ES Modules, it would be nice to be able to import the library in such a way. ESM support can be easily checked with the following command `npx is-esm "@geckos.io/server" 2.0.0-dev.2`.

**Comment by yandeu** [comment]
ESM is still not well supported on Node.js:

https://nodejs.org/docs/latest-v14.x/api/esm.html#esm_modules_ecmascript_modules

**Reply by panuchka** [reply]
Ah, yeah, in 14 it is not, but in 15 it is stable: https://nodejs.org/docs/latest-v15.x/api/esm.html#esm_modules_ecmascript_modules

Node 16 will go to 'current' status on the 20th of April this year and to 'active' 6 months later according to this https://nodejs.org/en/about/releases/

My point is that it will be stable pretty soon and I think it would ne a nice thing to utilize by all the library devs. I just wish to live in a world where we can import stuff easily with the same syntax in the client/server 😁

**Comment by antonrybalko** [comment]
```
import geckos from '@geckos.io/server'
const io = geckos.default()
```
Works fine

**Reply by yandeu** [reply]
On native Node.js v15? Without any build tools?

_I personally alway use TypeScript so I don't really care about native ESM support._

**Reply by antonrybalko** [reply]
On native Node.js v14.13.0
